<div>
    <h1>Productos en el carrito</h1>
    <h2>Carrito ID: {{cartId}}</h2>
    <ul>
        {{#each cart.products}}
        <li>
            <h3>{{this.id.titulo}}</h3>
            <p>Descripción: {{this.id.descripcion}}</p>
            <p>Precio: {{this.id.precio}}</p>
            <p>Thumbnail: {{this.id.thumbnail}}</p>
            <p>Categoría: {{this.id.categoria}}</p>
            <p>Código: {{this.id.code}}</p>
            <p>Stock: {{this.id.stock}}</p>
            <p>Disponible: {{this.id.disponible}}</p>
            <p>Cantidad: {{this.quantity}}</p>
            <button onclick="removeProduct('{{this.id._id}}', '{{../cartId}}')">Eliminar producto</button>
        </li>
        {{/each}}
    </ul>
    <button onclick="clearCart('{{cartId}}')">Eliminar todos los productos del carrito</button>
</div>

<script>
    async function clearCart(cartId) {
        try {
            const response = await fetch(`/api/carts/${cartId}`, {
                method: 'DELETE',
            });

            if (response.ok) {
                const result = await response.json();
                alert('Todos los productos han sido eliminados del carrito');
                location.reload();
            } else {
                const error = await response.json();
                alert(`Error: ${error.message}`);
            }
        } catch (error) {
            console.error('Error al eliminar todos los productos del carrito:', error);
            alert('Hubo un error al intentar eliminar todos los productos del carrito');
        }
    }

    async function removeProduct(productId, cartId) {
        console.log(`Cart ID: ${cartId}, Product ID: ${productId}`);
        try {
            const response = await fetch(`/api/carts/${cartId}/products/${productId}`, {
                method: 'DELETE',
            });

            if (response.ok) {
                const result = await response.json();
                alert('Producto eliminado del carrito');
                location.reload();
            } else {
                const error = await response.json();
                alert(`Error: ${error.message}`);
            }
        } catch (error) {
            console.error('Error al eliminar el producto del carrito:', error);
            alert('Hubo un error al intentar eliminar el producto del carrito');
        }
    }
</script>